api/controllers/v1/ncn.go

```diff
@@ -61,38 +61,71 @@ func NewNcnController(workflowService services.WorkflowService, logger utils.Log
 // @Security  OAuth2Application[admin]
 func (u NcnController) NcnCreateRebuildWorkflow(c *gin.Context) {
 	hostname := c.Param("hostname")
-	err := u.validator.ValidateHostname(hostname)
+	u.createRebuildWorkflow([]string{hostname}, c)
+}
+
+// NcnsCreateRebuildWorkflow
+// @Summary   End to end rolling rebuild ncns (workers only)
+// @Param     include  body  models.CreateRebuildWorkflowRequest  true  "hostnames to include"
+// @Tags      NCNs
+// @Accept    json
+// @Produce   json
+// @Success   200  {object}  models.Workflow
+// @Failure   400  {object}  utils.ResponseError
+// @Failure   404  {object}  utils.ResponseError
+// @Failure   500  {object}  utils.ResponseError
+// @Router    /v1/ncns/rebuild [post]
+// @Security  OAuth2Application[admin]
+func (u NcnController) NcnsCreateRebuildWorkflow(c *gin.Context) {
+	var requestBody models.CreateRebuildWorkflowRequest
+	if err := c.BindJSON(&requestBody); err != nil {
+		errResponse := utils.ResponseError{Message: fmt.Sprint(err)}
+		c.JSON(400, errResponse)
+		return
+	}
+	u.createRebuildWorkflow(requestBody.Hosts, c)
+
```

Comment:
Nit: extra newline, only noting this in one spot though not everywhere.

Reply:
linked to CASMPET-5466